jobs/cloud/worker.go

```diff
@@ -0,0 +1,284 @@
+}
+
+func (worker *Worker) ForgivenessCheck(userDifference int) bool {
+
+	systemValue, err := worker.app.Srv().Store.System().GetByName(model.OVER_USER_LIMIT_FORGIVEN_COUNT)
+
+	if err != nil {
+		mlog.Error("Error getting days over limit from system store", mlog.String("worker", worker.name), mlog.String("error", err.Error()))
+	}
+
+	var forgivenessCount int
+	if systemValue == nil {
+		forgivenessCount = 0
+	} else {
+		forgivenessCount, err = strconv.Atoi(systemValue.Value)
+		if err != nil {
+			forgivenessCount = 0
+		}
+	}
+
+	if forgivenessCount >= 3 {
+		// There's no forgiving someone who's broken our hearts 3 or more times
+		return false
+	}
+
+	forgivenessCount++
+	sysVar := &model.System{Name: model.OVER_USER_LIMIT_FORGIVEN_COUNT, Value: strconv.Itoa(forgivenessCount)}
+	err = worker.app.Srv().Store.System().SaveOrUpdate(sysVar)
+	if err != nil {
+		mlog.Error("Unable to save OVER_USER_LIMIT_FORGIVEN_COUNT", mlog.String("worker", worker.name), mlog.String("error", err.Error()))
+	}
+
+	_, err = worker.app.Srv().Store.System().PermanentDeleteByName(model.USER_LIMIT_OVERAGE_CYCLE_END_DATE)
+	if err != nil {
+		mlog.Error("Unable to reset USER_LIMIT_OVERAGE_CYCLE_END_DATE", mlog.String("worker", worker.name), mlog.String("error", err.Error()))
+	}
+
+	// Forgiven!
+	return true
+}
+
+func (worker *Worker) DoJob(job *model.Job) {
+	if claimed, err := worker.jobServer.ClaimJob(job); err != nil {
+		mlog.Warn("Worker experienced an error while trying to claim job",
+			mlog.String("worker", worker.name),
+			mlog.String("job_id", job.Id),
+			mlog.String("error", err.Error()))
+		return
+	} else if !claimed {
+		return
+	}
+
+	cloudUserLimit := *worker.app.Config().ExperimentalSettings.CloudUserLimit
+
+	var subscription *model.Subscription
+	var subErr *model.AppError
+	if worker.app.Srv().License() != nil && *worker.app.Srv().License().Features.Cloud {
+		subscription, subErr = worker.app.Cloud().GetSubscription()
+
+		if subErr != nil {
+			mlog.Error("Worker: Failed to retrieve subscription", mlog.String("worker", worker.name), mlog.String("job_id", job.Id), mlog.String("error", subErr.Error()))
+			worker.setJobError(job, subErr)
+			return
+		}
+
+		if subscription == nil {
+			mlog.Error("Worker: Failed to retrieve subscription", mlog.String("worker", worker.name), mlog.String("job_id", job.Id))
+			return
+		}
+	}
+
+	if subscription != nil && subscription.IsPaidTier == "true" {
+		mlog.Info("On Paid Tier, exiting", mlog.String("worker", worker.name))
+		return
+	}
+
+	count, err := worker.app.Srv().Store.User().Count(model.UserCountOptions{IncludeDeleted: false})
+
+	if err != nil {
+		mlog.Error("Worker: Failed to get active user count", mlog.String("worker", worker.name), mlog.String("job_id", job.Id), mlog.String("error", err.Error()))
+		return
+	}
+
+	userDifference := cloudUserLimit - count
+
+	systemValue, nErr := worker.app.Srv().Store.System().GetByName(model.USER_LIMIT_OVERAGE_CYCLE_END_DATE)
```

Comment:
Given that this is much later on, can we rename to what is actually has for readability?
```suggestion
	overageEndDateSystemValue, nErr := worker.app.Srv().Store.System().GetByName(model.USER_LIMIT_OVERAGE_CYCLE_END_DATE)
```
My suggestion is a bit wordy but just to convey the idea